core/robstoll-lib/atrium-core-robstoll-lib-common/src/main/kotlin/ch/tutteli/atrium/core/robstoll/lib/reporting/DetailedObjectFormatter.kt

```diff
@@ -41,7 +41,7 @@ abstract class DetailedObjectFormatterCommon(
      * @return The formatted [value].
      */
     @Suppress( /* TODO remove with 1.0.0 */ "DEPRECATION")
-    override fun format(value: Any?): String = when (value) {
+    override fun format(value: Any?): String = limitRepresentation(when (value) {
```

Comment:
I would like to see a different solution where one still sees what type the subject was and what identity hash the instance had. I think it would also make sense if we use an ellipsis. Say you have a very long list then the final result should be something like:
```
expected that subject: [1, 2, 3, 4, ...       (java.util.Arrays.ArrayList <1234789>)
```
and not
```
expected that subject: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 1
```
where the `...` in the first example shows that the output would have been longer.

Reply:
Oh sure. This certainly sounds more useful.
I get the example, but what do you mean exactly with ellipsis?

Reply:
suffix the representation with ` ...` in case it is longer than 10'000  (I am not a native speaker but I think `...` is called ellipsis in English :laughing:)

Reply:
Me neither but it seems plausible 😂 
But yea, I can do that

Reply:
I pushed my latest progress. If I still haven't got it right, I can change it again